### ISSUE TITLE
Arbitrary object primary key field

### ISSUE THREAD

**Issue by petrboros** [soynatan/django-easy-audit#29]
At this point, this package only supports IntegerField as PK field of audited objects. Is it possible to make the field more general. For instance, in our projects we use UUID fields as PK fields. Thank you for considering.

**Comment by jheld** [comment]
Yeah, I was wondering about this, too. We could use a `CharField` to open that up, though I do wonder if there is another way to do it.

**Comment by jheld** [comment]
One reason that IntegerField or CharField are nice is because we get to just reference the value by its value, not link it as a foreign/primary key. If the object gets deleted, we would lose our explicit reference, so we can't look up the object anymore. Just putting this in as background info.

**Comment by alan404** [comment]
This issue will break some third-party django app. For example, django rest token auth use a token model with the token as the primary. With this app token cannot be created or deleted.

**Comment by morlandi** [comment]
I do believe CharField is more appropriate than GenericForeignKey for auditing and logging purposes, even with the limitations  correctly described by @alan404.
Later on, you can still lookup the referenced object "manually", as long as it still exists.

**Comment by jheld** [comment]
Let's say we did use a generic foreign key. Docs: https://docs.djangoproject.com/en/2.2/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey

Seems we may be able to do it. Certain types can be coerced. If UUIDField can be, then this may work well!

We can make the object_id a CharField and then see if there are any tricks we can play with GenericForeignKey such that on_delete does nothing or something. If that works, then content_object would be an upgrade!

If we cannot, then I guess just CharField.
